src/config/ConfigManager.cpp

```diff
@@ -145,6 +145,67 @@ static void configHandleGapDestroy(void** data) {
         delete reinterpret_cast<CCssGapData*>(*data);
 }
 
+static int strToPangoWeight(const std::string& weight) {
+    auto loWeight{weight};
+    transform(weight.begin(), weight.end(), loWeight.begin(), ::tolower);
+
+    // values taken from Pango weight enums
+    if (loWeight == "thin")
+        return 100;
+    if (loWeight == "ultralight")
+        return 200;
+    if (loWeight == "light")
+        return 300;
+    if (loWeight == "semilight")
+        return 350;
+    if (loWeight == "book")
+        return 380;
+    if (loWeight == "normal")
+        return 400;
+    if (loWeight == "medium")
+        return 500;
+    if (loWeight == "semibold")
+        return 600;
+    if (loWeight == "bold")
+        return 700;
+    if (loWeight == "ultrabold")
+        return 800;
+    if (loWeight == "heavy")
+        return 900;
+    if (loWeight == "ultraheavy")
+        return 1000;
+
+    int w_i = std::stoi(weight);
+    if (w_i < 100 || w_i > 1000)
+        return 400;
+
+    return w_i;
+}
+
+static Hyprlang::CParseResult configHandleFontWeightSet(const char* VALUE, void** data) {
+    std::string V = VALUE;
+
+    if (!*data)
+        *data = new int;
+
+    const auto             DATA = reinterpret_cast<Hyprlang::INT*>(*data);
```

Comment:
this will not pass tests. custom types in hyprland must be a `ICustomConfigValueData` (see `SConfigOptionDescription::jsonify()`)

Reply:
before i go and make a CustomConfigValue class to represent "string or int", would it perhaps be more valuable to modify `CConfig::addConfigValue` in hyprlang to accept an optional handler function? this would allow more flexibility in how values are parsed, and/or allow `configStringToInt` to be decomposed into smaller, more precise parsing functions. on a side note, is there a reason why `Hyprlang::BOOL` doesn't exist?

Reply:
what? just make a new class inheriting the iface

Reply:
ok done. can you address my previous questions now? also, you mentioned that my change wouldn't "pass tests". is there an automated testing suite attached to this project? and if so, where is it and how do i run it? is this information in the wiki?

Reply:
it's in the works in #9297